src/frontend/apps/markdown/MarkdownEditor/index.tsx

```diff
@@ -0,0 +1,287 @@
+    );
+    return translation ? translation[content] : '';
+  };
+
+  const [localMarkdownContent, setLocalMarkdownContent] =
+    React.useState<string>('');
+  const [localTitle, setLocalTitle] = React.useState<string>('');
+  const [localIsDraft, setLocalIsDraft] = React.useState<boolean>(true);
+
+  const [contentChanged, setContentChanged] = React.useState(false);
+  const [renderedContent, setRenderedContent] = React.useState<string | null>(
+    null,
+  );
+  const [codemirrorView, setCodemirrorView] = React.useState<
+    EditorView | undefined
+  >();
+
+  const { mutate: saveDocumentTranslations, isLoading: isSaving } =
+    useSaveTranslations(markdownDocumentId, {
+      onSuccess: () => {
+        setContentChanged(false);
+        toast.success(intl.formatMessage(messages.saveSuccess));
+      },
+      onError: () => {
+        toast.error(intl.formatMessage(messages.saveFailure));
+      },
+    });
+
+  const canSaveDocument = () =>
+    !!localMarkdownContent && !!renderedContent && !!localTitle;
+  const saveDocument = () => {
+    if (canSaveDocument()) {
+      saveDocumentTranslations({
+        language_code: language,
+        title: localTitle,
+        content: localMarkdownContent,
+        rendered_content: renderedContent!,
+      });
+    }
+  };
+  const publishDocument = () => {
+    mutateDocument({
+      is_draft: false,
+    } as MarkdownDocument);
+    // This will invalidate query cache, and refetch all data
+  };
+
+  const updateEditorContent = (content: string) => {
+    if (!codemirrorView) return;
+    const update = codemirrorView.state.update({
+      changes: {
+        from: 0,
+        to: codemirrorView.state.doc.length,
+        insert: content,
+      },
+    });
+    codemirrorView.update([update]);
+    // Warning: this does not clear the state, ie. an "undo"
+    // in the editor will restore the previous content (bad
+    // when we are talking about language change).
+    // `codemirrorView.setState(state)` would be a better solution,
+    // but it would be a bad idea not to make it inside the
+    // `CodeMirrorEditor` component.
+  };
+
+  const onRenderedContentChange = (newRenderedContent: string | null) => {
+    setContentChanged(renderedContent !== newRenderedContent); // always true when using mermaid or latex...
+    setRenderedContent(newRenderedContent);
+  };
+  const onTitleChanged = (newTitle: string) => {
+    setContentChanged(true);
+    setLocalTitle(newTitle);
+  };
+
+  useEffect(() => {
+    queryClient.invalidateQueries(['markdown-documents', markdownDocumentId]);
+  }, [language]);
+
+  useEffect(() => {
+    if (!markdownDocument) return;
+    setLocalMarkdownContent(getTranslatedContent('content'));
+    setLocalTitle(getTranslatedContent('title'));
+    updateEditorContent(getTranslatedContent('content'));
+    setRenderedContent(getTranslatedContent('rendered_content'));
+    setLocalIsDraft(markdownDocument.is_draft);
+  }, [markdownDocument, language]);
```

Comment:
```suggestion
  }, [markdownDocument, getTranslatedContent]);
```

I don't see where `language` is used in this `useEffect`. You should also add `getTranslatedContent` in the dependency array

Reply:
The `language` is used inside `getTranslatedContent`, I wasn't aware we can put function in here, that's why I was using the changing value instead. I will do like you suggest :)